timesketch/lib/analyzers/hashr_lookup.py

```diff
@@ -0,0 +1,303 @@
+        db_string_redacted = (f'postgresql://{db_user}:***@'
+                              f'{db_address}:{db_port}/{db_name}')
+        db = sqla.create_engine(
+            db_string, connect_args={'connect_timeout': 10})
+        try:
+            db.connect()
+            logger.info('Successful connected to hashR postgress database: %s',
+                        db_string_redacted)
+            self.hashr_conn = db
+            return True
+        except sqla.exc.OperationalError as err:
+            logger.error(
+                '!!! Connection to the hashR postgres database not '
+                'possible! -- Provided connection string: "%s"'
+                ' -- Error message: %s', db_string_redacted, str(err))
+            return ('Connection to the database FAILED. Please check the celery'
+                    ' logs and make sure you have provided the correct database'
+                    ' information in the analyzer file!')
+
+    def batch_hashes(self, hash_list, batch_size=DEFAULT_BATCH_SIZE):
+        """ Generator function for slicing the hash list into batches
+
+        Args:
+          hash_list: A list of hash values.
+          batch_size: Size of each batch. Defalt defined in class var.
+        """
+        for i in range(0, len(hash_list), batch_size):
+            yield hash_list[i : i + batch_size]
+
+    def check_against_hashr(self, sample_hashes):
+        """Check a list of hashes against the hashR database.
+
+        Args:
+          sample_hashes:  A list of hash values that shall be checked
+                          against the hashR database.
+
+        Returns:
+          matching_hashes:  A dict containing all hashes that are found in the
+                            hashR database. (If add_source_attribute is True
+                            it also contains the corresponding source images.)
+          None:             If there are no matches, the function returns None.
+
+        Raises:
+          Exception:  Raises an exception if the provided sample_hashes
+                      parameter is not of type list or dict.
+        """
+        if not isinstance(sample_hashes, list):
+            raise Exception('The check_against_hashR function only accepts '
+                            'type<list> as input. But type '
+                            f'{type(sample_hashes)} was provided!')
+
+        meta_data = sqla.MetaData(bind=self.hashr_conn)
+        sqla.MetaData.reflect(meta_data)
+        samples_table = meta_data.tables['samples']
+        sources_table = meta_data.tables['sources']
+        samples_sources_table = meta_data.tables['samples_sources']
+
+        matching_hashes = {}
+
+        with self.hashr_conn.connect() as conn:
+            batch_counter = 1
+            total_batches = ceil(len(sample_hashes)/self.query_batch_size)
+            for batch in self.batch_hashes(sample_hashes, self.query_batch_size):
+                logger.info('Processing %d/%d batches...',
```

Comment:
Legit point. I'm a fan of verbose output of logs and code. But with the use-case of this analyzer it is not necessary to have this on a info level since no user will ever see it. I will check which are relevant as info and everything else goes into debug level.

Reply:
So I changed all info events to debug events, except the return message at the end. If someone watches the celery console output, they are probably debugging anyway and can activate the debug level.